id.go

```diff
@@ -274,3 +278,8 @@ func (id *ID) Scan(value interface{}) (err error) {
 		return fmt.Errorf("xid: scanning unsupported type: %T", value)
 	}
 }
+
+// IsNil Returns true if this is a "nil" ID
+func (id ID) IsNil() bool {
```

Comment:
What would be the use-case for this?

Reply:
As an example if you use these ID as PK and expose them for a REST API.
you would use FromString(QueryStringID) to get the xid.ID
In the case where someone would submit an empty string, this IsNil allows to check this.

Arguably, we can also simply compare with the FromString(""), but as this returns an error, it needs to be decomposed into 2 steps:
```
emptyID, _ := FromString("")
if id == emptyID
```

instead of
```
if id.IsNil()
```

Reply:
Ok